### PR TITLE
opencv_4.4.0.imx: add recipe

### PR DESCRIPTION
**Pull request by chrisdimich**
Add recipe to support additional machine learning demos and tiny dnn.

### CODE REVIEW

**Review by otavio — CHANGES_REQUESTED** [review]
Awesome, however we need to add the preferences and compatibility here otherwise it'll have multiple providers for it.

**Comment by chrisdimich** [comment]
Right, as we discussed I was going to add that to imx-base.inc, but was waiting for this to be pushed first. Let me add that to this PR.

File: recipes-support/opencv/opencv_4.4.0.imx.bb [hunk 1]
```diff
@@ -0,0 +1,295 @@
...
+    install -d ${D}${datadir}/OpenCV/samples/bin/
+    cp -f bin/example_* ${D}${datadir}/OpenCV/samples/bin/
+    if ${@bb.utils.contains('PACKAGECONFIG', 'test', 'true', 'false', d)}; then
+        cp -r share/opencv4/testdata/cv/face/* ${D}${datadir}/opencv4/testdata/cv/face/
+    fi
+}
+
+FILES_${PN}-samples += "${datadir}/OpenCV/samples"
+
+########## End of i.MX overrides ##########
```
Comment (otavio) [inline recipes-support/opencv/opencv_4.4.0.imx.bb:293]: We need to restrict the recipe to mx8.

**Comment by chrisdimich** [comment]
@otavio gentle reminder :) I made the change you requested and restricted recipe to mx8 only.

File: recipes-support/opencv/opencv_4.4.0.imx.bb (hunk 1 above)
Comment (otavio) [reply recipes-support/opencv/opencv_4.4.0.imx.bb:295]: This commit belongs to the recipe patch, not the `imx-base.inc`.

**Comment by chrisdimich** [comment]
@otavio I split the commits. Is this all good now?

**Review by otavio — APPROVED** [review]

**Pull request merged**